python_modules/dagster/dagster/_core/definitions/sensor_definition.py

```diff
@@ -301,6 +317,11 @@ def advance_all_cursors(self):
         materializations_by_key = self.latest_materialization_records_by_key()
         self.advance_cursor(materializations_by_key)
 
+    @public  # type: ignore
+    @property
+    def assets_by_key(self) -> Mapping[AssetKey, "AssetsDefinition"]:
```

Comment:
I think this one is also worth changing. Alternatively, we could just expect users to do `context.repository_def.assets_defs_by_key`